typescript/05/main.ts

```diff
@@ -0,0 +1,45 @@
+type Subjects = 'math' | 'geo' | 'english';
+type Grades = 1 | 2 | 3 | 4 | 5;
+
+interface Student {
+	firstName: string;
+	lastName: string;
+	subjectsAndGrades: { [subjects: string]: Grades };
+}
+
+function Student(this: Student, firstName: string, lastName: string) {
+	this.firstName = firstName;
+	this.lastName = lastName;
+	this.subjectsAndGrades = {};
+}
+
+Student.prototype.addGrade = function (subject: Subjects, grade: Grades) {
+	if (typeof this.subjectsAndGrades[subject] === 'undefined') {
+		this.subjectsAndGrades[subject] = [];
+	}
+	this.subjectsAndGrades[subject].push(grade);
+};
+Student.prototype.getAvgGrade = function (subject: Subjects) {
+	if (typeof subject === 'undefined') {
+		const arrayWithGrades = Object.values(this.subjectsAndGrades).reduceRight((acc: number, curr: number) => accumulator.concat(currentValue));
```

Comment:
Czy faktycznie kompilował Ci się ten kod? U mnie nie działa ;P

Reply:
Poprawiłem, faktycznie był błąd w kodzie, już jest okej

Reply:
Musisz na to bardzo uważać. Jeśli na rekrutacji zrobiłbyś taką gafę to nie sądzę, aby ktoś się nad tym pochylał ;)